### PR TITLE
Cover support and battery support for Aqara roller E1 (lumi.curtain.acn002)

### PR DESCRIPTION
**Pull request by schwickster**
Haven't gotten far. But the quirk is idenified and applied for the new Aqara roller E1.
https://github.com/zigpy/zha-device-handlers/issues/1181

Still doesn't fix the weird light toggle or going down.

### CODE REVIEW

**Comment by schwickster** [comment]
For now I'll throw in my tower in the ring. These changes bring the following:
- Quirk identifies the roller shade from Aqara
- Fixes the device being identified as a light
- Removes a switch that doesn't do anything
- Added state to cluster which shows if the device is charging or not

Still todo:
- Add battery state (Adding XiaomiPowerConfiguration reports unknown in Home Assistant)
- Remap current_position_lift_percentage to AnalogOutput > present_value.
- Fix run test in the checks

**Comment by schwickster** [comment]
Any chance of accepting this PR?

**Comment by dmulcahey** [comment]
It’s closed 🤔

**Comment by schwickster** [comment]
> It’s closed 🤔

Apologies, thought that was the procedure for getting it accepted. I'll open it.

**Comment by schwickster** [comment]
I fixed the unit test. Newbie mistake of mine...
It's ready to be merged now (in case of approval)

File: zhaquirks/xiaomi/__init__.py [hunk 1]
```diff
@@ -44,6 +44,7 @@
     VALUE,
     ZHA_SEND_EVENT,
 )
+from zhaquirks.xbee import uint8_t
```
Comment (dmulcahey) [inline zhaquirks/xiaomi/__init__.py:47]: Please import this from zigpy

Comment (schwickster) [reply zhaquirks/xiaomi/__init__.py]: Fixed

File: zhaquirks/xiaomi/aqara/roller_curtain_e1.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+
+class RollerE1AQ(XiaomiCustomDevice):
+    """Aqara Roller Shade Driver E1 device."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+
```
Comment (dmulcahey) [inline zhaquirks/xiaomi/aqara/roller_curtain_e1.py:42]: can you add the simple descriptor comment to this. Look at other quirks to see an example. You can get this by removing the device completely from HA and then pairing it again.

**Review by dmulcahey — CHANGES_REQUESTED** [review]

File: zhaquirks/xiaomi/aqara/roller_curtain_e1.py (hunk 2 above)
Comment (schwickster) [reply zhaquirks/xiaomi/aqara/roller_curtain_e1.py:42]: You mean add it to the signature?
Like this https://github.com/zigpy/zha-device-handlers/blob/79430799ddd887b416f5cca47361df6715add4e1/zhaquirks/xiaomi/__init__.py#L85
and this https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/xiaomi/aqara/vibration_aq1.py#L158

Comment (dmulcahey) [reply zhaquirks/xiaomi/aqara/roller_curtain_e1.py:42]: Nope. Like this: < https://github.com/zigpy/zha-device-handlers/blob/71dd9e5167ea6f786301a66d9e9bec4b9d1f6032/zhaquirks/xiaomi/aqara/illumination.py#L28>

I have to go back and update the others

Comment (schwickster) [reply zhaquirks/xiaomi/aqara/roller_curtain_e1.py:42]: OK thanks. Will add it later today or tomorrow.

Comment (schwickster) [reply zhaquirks/xiaomi/aqara/roller_curtain_e1.py:42]: Done!

**Comment by schwickster** [comment]
As reported in https://github.com/zigpy/zha-device-handlers/issues/1181 the cover entity is now working.  This was the biggest bug for this device.

Still todo:
- Figure out what to do with analog output
- Battery reporting

I'll try looking into the battery soon. This PR can be merged (if no further reviews). I'll create a seperate PR for future fixes/changes.

**Comment by schwickster** [comment]
As reported in https://github.com/zigpy/zha-device-handlers/issues/1181. Battery readings are now working. I think issue https://github.com/zigpy/zha-device-handlers/issues/1181 can be resolved now.

**Comment by dmulcahey** [comment]
I am going to merge this as is. There are things we can do to clean this up in the future.

**Comment by schwickster** [comment]
> I am going to merge this as is. There are things we can do to clean this up in the future.

Just to be curious and learn from mistakes, what things can be done better? Thanks for the merge!

**Pull request merged**